ee/tabby-webserver/src/service/job.rs

```diff
@@ -45,12 +46,37 @@ impl JobService for DbConn {
         let (limit, skip_id, backwards) = graphql_pagination_to_filter(after, before, first, last)?;
         let rowids = ids.map(|ids| ids.into_iter().filter_map(|x| x.as_rowid().ok()).collect());
         Ok(self
-            .list_job_runs_with_filter(rowids, limit, skip_id, backwards)
+            .list_job_runs_with_filter(rowids, jobs, limit, skip_id, backwards)
             .await?
             .into_iter()
             .map(Into::into)
             .collect())
     }
+
+    async fn compute_job_run_stats(&self, jobs: Option<Vec<String>>) -> Result<JobStats> {
+        let jobs = self
+            .list_job_runs(None, jobs, None, None, None, None)
+            .await?;
+        let mut stats = JobStats {
+            success: 0,
+            failed: 0,
+            pending: 0,
+        };
+
+        for job in jobs {
```

Comment:
should this be a aggregation query (`SELECT sum(success), ...`) in sql?

Reply:
No, there isn't a good way to do this - it was the first thing I thought of, but it would basically end up needing to be broken up into three separate aggregation queries, since each query has its own condition. I believe that would be much slower than just querying once and performing the aggregation ourselves.

Reply:
i don't get it - this query doesn't come with it's condition. so query is effectively as below:

```
SELECT sum(exit_code == 0), sum(exit_code == 1), sum(exit_code == null) FROM job_runs WHERE [job = :job]
```

Reply:
I didn't think of that, and now that I'm looking at it, I realize it's not even a special syntax but just a consequence of booleans being treated as ints